core/src/main/java/org/apache/iceberg/SnapshotProducer.java

```diff
@@ -351,6 +351,7 @@ private Map<String, String> summary(TableMetadata previous) {
         SnapshotSummary.ADDED_EQ_DELETES_PROP,
         SnapshotSummary.REMOVED_EQ_DELETES_PROP);
 
+    builder.putAll(EnvironmentContext.get());
```

Comment:
can you add a simple test to `TestSnapshotSummary` to indicate that `iceberg-version` is part of the produced summary map